File: leetcode/2_two_pointers/977_Squares_of_a_Sorted_Array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left, right, sq = 0, len(nums) - 1, 0
        res = [0] * len(nums)
        for i in range(len(nums) - 1, -1, -1):
            if abs(nums[left]) < abs(nums[right]):
                sq = nums[right]
                right -= 1
            else:
                sq = nums[left]
                left += 1
            res[i] = sq * sq
        return res

File: leetcode/2_two_pointers/977_Squares_of_a_Sorted_Array.go
import "math"

func sortedSquares(nums []int) []int {
	left, right := 0, len(nums)-1
	res := make([]int, len(nums))
	var sq int
	for i := len(nums) - 1; i >= 0; i-- {
		if math.Abs(float64(nums[left])) < math.Abs(float64(nums[right])) {
			sq = nums[right]
			right--
		} else {
			sq = nums[left]
			left++
		}
		res[i] = sq * sq
	}
	return res
}
